vcd/resource_vcd_org_vdc_access_control.go

```diff
@@ -0,0 +1,236 @@
+	if err != nil {
+		return diag.Errorf("error when setting VDC control access parameters - %s", err)
+	}
+
+	d.SetId(vdc.Vdc.ID)
+	return resourceVdcVdcAccessControlRead(ctx, d, meta)
+}
+
+func resourceVdcVdcAccessControlRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("error while reading Org - %s", err)
+	}
+
+	vdc, err := org.GetVDCById(d.Id(), false)
+	if err != nil {
+		if govcd.IsNotFound(err) {
+			d.SetId("")
+		} else {
+			return diag.Errorf("error while reading VDC - %s", err)
+		}
+	}
+
+	controlAccessParams, err := vdc.GetControlAccess(true)
+	if err != nil {
+		return diag.Errorf("error getting control access parameters - %s", err)
+	}
+
+	dSet(d, "shared_with_everyone", controlAccessParams.IsSharedToEveryone)
+	if controlAccessParams.EveryoneAccessLevel != nil {
+		dSet(d, "everyone_access_level", *controlAccessParams.EveryoneAccessLevel)
+	}
+
+	if controlAccessParams.AccessSettings != nil {
+		accessControlListSet, err := accessControlListToSharedSet(controlAccessParams.AccessSettings.AccessSetting)
+		if err != nil {
+			return diag.Errorf("error converting slice AccessSetting into set - %s", err)
+		}
+
+		err = d.Set("shared_with", accessControlListSet)
+		if err != nil {
+			return diag.Errorf("error setting shared_with attribute - %s", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceVdcVdcAccessControlDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// When deleting VDC access control, VDC won't be share with anyone, neither everyone not any single user/group
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		diag.Errorf("error when retrieving VDC - %s", err)
+	}
+
+	controlAccessParams := &types.ControlAccessParams{
+		IsSharedToEveryone: false,
+	}
+
+	_, err = vdc.SetControlAccess(controlAccessParams, true)
+	if err != nil {
+		return diag.Errorf("error when setting VDC control access parameters - %s", err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceVdcVdcAccessControlImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 2 {
+		return nil, fmt.Errorf("resource name must be specified as org.catalog")
+	}
+
+	orgName, vdcName := resourceURI[0], resourceURI[1]
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdc(orgName, vdcName)
+	if err != nil {
+		return nil, fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	accessControlParams, err := vdc.GetControlAccess(true)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't retrieve access control parameters - %s", err)
+	}
+
+	dSet(d, "org", orgName)
+	dSet(d, "vdc", vdcName)
+	dSet(d, "shared_with_everyone", accessControlParams.IsSharedToEveryone)
```

Comment:
If I remember correctly, you don't need set everything here, as after import automatically `read` is called.

Reply:
Yes - just set the ID and `org`, `vdc` fields (because the resource itself does not set it) - everything will be read by Terraform and less clutter here.